### ISSUE TITLE
Make an import function

### ISSUE THREAD

**Issue by Polsander** [ferncabrera/open_ims#212]
Gave this some thought.

We need an import data function. (Button can be next to the export one).

The idea is that our current creation process for customers, vendors, and purchase orders only allows us to create one entity or database row at a time.  (i.e., if I want to make 5 new customers to set up my business of customers which I already have, then I will have to go through that big form field multiple times). 

Instead, what we can do is give the user the option to upload an excel spreadsheet, which we can asynchronously handle and use the rows there to populate multiple users, customers, whatever!

**Proposed Idea of how this will work:**

1. The import button will open a modal. The modal will show 2 options. 
2. The first option is a button that downloads an excel template file to the user's computer. This template file will contain the boiler plate layout required by the user for uploading their excel file. (i.e. all headers, required fields, etc.) As well as instructions on what is allowed and not allowed in the excel file (characters, length, etc.).
3. The second option is an upload function. The user fills out the template with however many rows of data they want, and then upload it.
4. Once uploaded, the user clicks submit and then open_ims begins to handle the creation of data **ASYNCHRONOUSLY** (because we don't know how much data the user will upload, maybe we want to limit it?).
5. Once the upload is complete, the user is then notified and can see changes on the database or table.
6. If upload fails, user is notified with an error code and message hinting as to what failed.

**Main issues**
- I do not have much experience in dealing with asynchronous systems that work in the background in terms of development. I have used node-cron time scheduler before, but aside from that, I do not know what would be appropriate for this case.
- Need to gauge how critical or important this is. Backlog ticket, or would we move it to critical? We can always populate Austin's database in person using SQL, but when we ship this out as open source, we should let users be able to handle that themselves. Just depends whether or not you think Austin should have this or not.
- Where do we notify upload completions/failures? May need to work on implementing functionality on the little user bell icon on the bottom left.
- Is there a better way of implementing this besides my proposed method?

All good things to think about. Please let me know @ferncabrera and how you would like to proceed.
Will assign this ticket to us both for now. Eventually when this is worked on, we can split it into smaller tickets for the front-end modals, templating, notification showings, and upload functionality and checks.

**Comment by Polsander** [comment]
Have individual tasks that cover and will make this a reality.
